### PR TITLE
Modify data pipeline to accept compile parameters

### LINKED ISSUE #7: Compilation parameters (data pipeline)
- [x] Add compilation parameters to local state in `components/Editor.js`
- [x] Update redux action `runRequest` to take in object of parameters in addition to code
- [x] Update all `runRequest` calls to take in new parameters
- [x] Depends on https://github.com/cs121-team-panda/coconut-interpreter-react/issues/6

### Reference
- Compilation parameters: http://coconut.readthedocs.io/en/master/DOCS.html#setup

### PR DESCRIPTION
**Pull request by nkadonaga**
Resolve #7 

- [x] circleCI tests pass
- [x] frontend tests pass
- [x] empty compile params works
(just clicked Run on default code here)
screen shot 2018-04-01 at 7 11 18 pm

- [x] nonempty compile params works
(used Dispatch to specify the args here)
screen shot 2018-04-01 at 7 11 40 pm

### CODE REVIEW

File: src/components/Editor.js [hunk 1]
```diff
@@ -45,12 +47,13 @@ export default class Editor extends Component<Props, State> {
     return errorMarker(code, errorLine, errorCall, styles.errorMarker);
   };
 
-  handleChange = (newCode: string) => {
-    this.setState({ code: newCode });
+  handleChange = (newCode: string, newArgs: string) => {
```
Comment (jcruz) [inline src/components/Editor.js]: `handleChange` updates the state when the user types. No need to update args here.

File: src/components/Editor.js [hunk 2]
```diff
@@ -17,19 +17,21 @@ import {
...
-  runRequest: (code: string) => void,
+  runRequest: (code: string, args: string) => void,
   loading: boolean,
   errorLine: ?number,
   errorCall: ?string,
 };
 
 type State = {
   code: string,
+  args: string,
```
Comment (jcruz) [inline src/components/Editor.js]: Type of `args` is probably an object not a string. We should define a type for `args` somewhere, probably in `actions.js`.
```javascript
type Args = {
  target: string,
  ...
}
```
and then we can change this to
```
args: Args
```

**Review by jcruz — COMMENTED** [review]
Noice!

File: src/components/Editor.js (hunk 1 above)
Comment (nkadonaga) [reply src/components/Editor.js]: Cherry-picked [c328c62](https://github.com/cs121-team-panda/coconut-interpreter-react/pull/11/commits/c328c62a57f28acffd03891e1183ceed23cb5743)

File: src/components/Editor.js (hunk 2 above)
Comment (nkadonaga) [reply src/components/Editor.js]: Created 'Args' type, but I still need to remove all parameters except python target (#18), should I start that in a new PR?

**Review by jcruz — APPROVED** [review]
LGTM

**Pull request merged**